code/modules/vehicles/quad.dm

```diff
@@ -49,12 +57,16 @@
 
 /obj/vehicle/train/engine/quadbike/Moved(atom/old_loc, direction, forced = FALSE)
 	. = ..() //Move it move it, so we can test it test it.
-	if(!istype(loc, old_loc.type) && !istype(old_loc, loc.type))	//Did we move at all, and are we changing turf types?
+	get_turf_speeds(old_loc)
+	handle_vehicle_icon()
+
+/obj/vehicle/train/engine/quadbike/proc/get_turf_speeds(atom/prev_loc)
+	if(!istype(loc, prev_loc.type) && !istype(prev_loc, loc.type))	//Did we move at all, and are we changing turf types?
```

Comment:
```suggestion
	// Same speed if turf type doesn't change
	if(istype(loc, prev_loc.type) || istype(prev_loc, loc.type)
		return
```
And then dedent the rest of this proc.
Low-key wishing there was some way to attach this to the turf type but then snowmobiles are supposed to do better on snow so that's ultimately tied to the mover more than the terrain